### PR TITLE
#967 gantt wpcolors bystage

### LINKED ISSUE #967: Gantt - Color WPs by Stage
### Description

https://docs.google.com/spreadsheets/d/1KMaX7BX4c7TXwEU0zSwl87XUk_U6zp7as5WFV6ToyFM/edit#gid=0

on their gantt chart, its colored by stage

### Acceptance Criteria

wp's on the gantt should be colored by their stage

### Proposed Solution

1. make a new file called enum-pipes.ts in the frontend utils folder. move the textMap and colorMap from workPackageStageChip there and rename them to workPackageStageTextPipe and workPackageStageColorPipe (make sure the old file now imports these and uses them properly)
2. now use these pipes and color each work package accordingly. if a wp has no stage, use the default grey we're using now
3. for progress, make the filled in part darker and the not finished part lighter (like if wp is 33% progress, the first third will be darker and the next two thirds will be lighter). this might mean we need to do different colors than defined before but we'll see, could also just involve messing with the opacity of those colors

this will probably take some trial and error so check in often

### Mocks

https://docs.google.com/spreadsheets/d/1KMaX7BX4c7TXwEU0zSwl87XUk_U6zp7as5WFV6ToyFM/edit#gid=0

### PR DESCRIPTION
**Pull request by ibarra000**
## Changes
- Created enum-pipes.ts
- moved workPackageStageTextPipe & workPackageStageColorPipe into enum-pipes.ts
- Updated WorkPackageStageChip to use workPackageStageTextPipe & workPackageStageColorPipe 
- WorkPackages Colored by Stage on Gantt

## Notes
Peyton carried 

## Screenshots
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #967

### CODE REVIEW

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
awesome work!!

File: src/frontend/src/components/WorkPackageStageChip.tsx [hunk 1]
```diff
@@ -4,34 +4,24 @@
...
-  const color: string = colorMap[stage];
-
-  const text: string = textMap[stage];
+  let color: string;
+  let text: string;
+
+  if (stage) {
+    color = WorkPackageStageColorPipe[stage];
+    text = WorkPackageStageTextPipe[stage];
+  } else {
```
[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply src/frontend/src/components/WorkPackageStageChip.tsx:21]: you cant make it take undefined, a record can only be a string a number or a symbol

[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply src/frontend/src/components/WorkPackageStageChip.tsx:21]: we could use a ?? and pass in like “” and then check for that, so then the record would be | string

File: src/frontend/src/pages/GanttPage/GanttPackage/components/task-item/bar/bar-display.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ type BarDisplayProps = {
   progressWidth: number;
   barCornerRadius: number;
   styles: {
-    backgroundColor: string;
+    backgroundColor?: string;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/components/task-item/bar/bar-display.tsx:15]: this isnt optional

File: src/frontend/src/pages/GanttPage/GanttPackage/types/bar-task.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export interface BarTask extends Task {
   handleWidth: number;
   barChildren: BarTask[];
   styles: {
-    backgroundColor: string;
+    backgroundColor?: string;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/types/bar-task.ts:16]: same thing here i think?

File: src/frontend/src/utils/enum-pipes.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { yellow, green, blue, purple, grey } from '@mui/material/colors';
+import { WorkPackageStage } from 'shared';
+
+// maps stage to the desired color
+export function WorkPackageStageColorPipe(stage: WorkPackageStage | undefined): string {
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/enum-pipes.ts:10]: we dont really use function syntax we say export const WorkPacka... = (stage: Wo...) => {
...
}

File: src/frontend/src/utils/enum-pipes.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      return blue[600];
+    case WorkPackageStage.Integration:
+      return purple[400];
+    case undefined:
+      return grey[500];
+  }
+}
+
+// maps stage to the desired badge display text
+export function WorkPackageStageTextPipe(stage: WorkPackageStage | undefined): string {
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/enum-pipes.ts:26]: same thing as above

File: src/frontend/src/utils/gantt.utils.ts [hunk 6]
```diff
@@ -112,6 +116,9 @@ export const transformProjectToGanttTask = (project: Project, expanded: boolean)
     type: 'project',
     hideChildren: !expanded,
     teamName,
+    styles: {
```
[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply src/frontend/src/utils/gantt.utils.ts:119]: i think u still have to address this.

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Good job so far still a couple things i think im gonna have anthony look at this too.

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts [hunk 7]
```diff
@@ -33,12 +29,12 @@ export const convertToBarTasks = (
       barCornerRadius,
       handleWidth,
       rtl,
-      barProgressColor,
-      barProgressSelectedColor,
+      t.styles?.backgroundColor ?? '#9d9d9d',
```
Comment (anthonybernardi) [inline src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:32]: i dont think this is a great place for the defaults to be since no other defaults are in this file

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts [hunk 8]
```diff
@@ -33,12 +29,12 @@ export const convertToBarTasks = (
...
+      t.styles?.backgroundColor ?? '#9d9d9d',
       barBackgroundColor,
       barBackgroundSelectedColor,
-      projectProgressColor,
-      projectProgressSelectedColor,
+      t.styles?.backgroundColor ?? '#e50000',
```
Comment (anthonybernardi) [inline src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:36]: same thing

File: src/frontend/src/utils/enum-pipes.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  switch (stage) {
+    case WorkPackageStage.Research:
+      return yellow[900];
+    case WorkPackageStage.Design:
+      return green[600];
+    case WorkPackageStage.Manufacturing:
+      return blue[600];
+    case WorkPackageStage.Integration:
+      return purple[400];
+    case undefined:
```
Comment (anthonybernardi) [inline src/frontend/src/utils/enum-pipes.ts:20]: can this be default instead

File: src/frontend/src/utils/enum-pipes.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  switch (stage) {
+    case WorkPackageStage.Research:
+      return 'Research';
+    case WorkPackageStage.Design:
+      return 'Design';
+    case WorkPackageStage.Manufacturing:
+      return 'Manufacturing';
+    case WorkPackageStage.Integration:
+      return 'Integration';
+    case undefined:
```
Comment (anthonybernardi) [inline src/frontend/src/utils/enum-pipes.ts:36]: same thign with default

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx [hunk 11]
```diff
@@ -59,7 +59,7 @@ const WorkPackageSummary: React.FC<WorkPackageSummaryProps> = ({ workPackage })
             {workPackage.name}
           </Link>
         </Box>
-        {workPackage.stage ? <WorkPackageStageChip stage={workPackage.stage} /> : null}
+        <WorkPackageStageChip stage={workPackage.stage} />
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx:62]: if it has no chip we dont want it to show up at all so the previous code was right (this way it'll show up as ag rey chip saying "none")

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageDetails.tsx [hunk 12]
```diff
@@ -21,7 +21,7 @@ const WorkPackageDetails: React.FC<WorkPackageDetailsProps> = ({ workPackage })
       title={'Work Package Details'}
       headerRight={
         <>
-          {workPackage.stage ? <WorkPackageStageChip stage={workPackage.stage} /> : null}
+          <WorkPackageStageChip stage={workPackage.stage} />
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageDetails.tsx:24]: same thing as in the work package summary

File: src/frontend/src/pages/GanttPage/GanttPackage/types/bar-task.ts (hunk 3 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/GanttPage/GanttPackage/types/bar-task.ts:16]: ye

**Review by anthonybernardi — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts (hunk 7 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:32]: this is the only place we map individual styles for tasks

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx (hunk 11 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx:62]: are you sure? I think none would actually be better imo?

Comment (Peyton-McKee) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx:62]: Per slack just change the default case in the text wp stage function to say “No Stage”

File: src/frontend/src/pages/GanttPage/GanttPackage/components/gantt/gantt.tsx [hunk 13]
```diff
@@ -117,11 +117,7 @@ export const Gantt: React.FunctionComponent<GanttProps> = ({
         barBackgroundColor,
         barBackgroundSelectedColor,
         projectProgressColor,
-        projectProgressSelectedColor,
-        projectBackgroundColor,
-        projectBackgroundSelectedColor,
-        milestoneBackgroundColor,
-        milestoneBackgroundSelectedColor
+        projectProgressSelectedColor
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/components/gantt/gantt.tsx:120]: undo these changes

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts [hunk 14]
```diff
@@ -11,12 +11,8 @@ export const convertToBarTasks = (
   barCornerRadius: number,
   handleWidth: number,
   rtl: boolean,
-  barProgressColor: string,
-  barProgressSelectedColor: string,
   barBackgroundColor: string,
   barBackgroundSelectedColor: string,
-  projectProgressColor: string,
-  projectProgressSelectedColor: string,
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:19]: undo the above four changes

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts (hunk 7 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:32]: make the default be the barProgressColor

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts [hunk 15]
```diff
@@ -33,12 +29,12 @@ export const convertToBarTasks = (
       barCornerRadius,
       handleWidth,
       rtl,
-      barProgressColor,
-      barProgressSelectedColor,
+      t.styles?.backgroundColor ?? '#9d9d9d',
+      t.styles?.backgroundColor ?? '#9d9d9d',
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:33]: make the default be the barProgressSelectedColor

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts (hunk 8 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:36]: make the default be projectProgressColor

File: src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts [hunk 16]
```diff
@@ -33,12 +29,12 @@ export const convertToBarTasks = (
...
       barBackgroundColor,
       barBackgroundSelectedColor,
-      projectProgressColor,
-      projectProgressSelectedColor,
+      t.styles?.backgroundColor ?? '#e50000',
+      t.styles?.backgroundColor ?? '#e50000',
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/GanttPage/GanttPackage/helpers/bar-helper.ts:37]: make default be barprogressselectedcolor

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
building off of anthonys comments and slack discussion

**Review by Peyton-McKee — APPROVED** [review]
yeah this looks good, this is kinda how we have to do this ig

**Review by anthonybernardi — APPROVED** [review]
nice!
